File: code/bit_manipulation/src/convert_number_binary/ConvertNumberBinary.java
// Part of Cosmos by OpenGenus Foundation
public class ConvertNumberBinary {
	public static void main(String[] args) {
		String binary = toBinary(20);
		System.out.println(binary);

		int number = toNumber("10101");
		System.out.println(number);
	}

	public static String toBinary(int n) {
		StringBuilder binary = new StringBuilder("");
		while (n > 0) {
			if ((n&1) == 0)
				binary.append('0');
			else
				binary.append('1');

			n >>= 1;
		}
		return binary.toString();
	}

	public static int toNumber(String s) {
		int number = 0;
		int n = s.length();
		for (int i = 0; i < n; i++) {
			if (s.charAt(i) == '1') {
				number += (1 << (n - 1 - i));
			}
		}
		return number;
	}
}

File: code/bit_manipulation/src/convert_number_binary/convert_number_binary.js
/* Part of Cosmos by OpenGenus Foundation */

//Using built in functions:

function toBinary(val) {
  return val.toString(2);
}

function fromBinary(bitString) {
  return parseInt(bitString, 2);
}

//Using bitshifts

function toBinary_B(val) {
  let out = "";
  while (val > 0) {
    out = (val & 1) + out;
    val >>= 1;
  }
  return out;
}

function fromBinary_B(bitString) {
  let out = 0;
  let l = bitString.length;
  let i = 0;
  for (; i < l; i++) {
    out += +bitString[i] ? 1 << (l - i - 1) : 0;
  }
  return out;
}
